packages/camera/camera_avfoundation/ios/Classes/CameraPlugin.m

```diff
@@ -56,6 +55,10 @@ - (void)initDeviceEventMethodChannel {
       [[FLTThreadSafeMethodChannel alloc] initWithMethodChannel:methodChannel];
 }
 
+- (void)detachFromEngineForRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
```

Comment:
I did not clear out `inProgressSavePhotoDelegates` since the delegate does not hold strong reference to its owner anymore (deleted the corresponding comments saying so)